src/common/Utils.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include "../Exceptions.h"
+
+template <typename Func, typename... Args>
+int covered_call(int error_value, Func func, Args... args)
```

Comment:
nice, great wrapper,
for a small project this will work though for large-scale project this will make it hard for the eyes to read